sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLPlannerImpl.java

```diff
@@ -171,6 +178,18 @@ public RelRoot rel(String sql, QueryParameters params) {
                 SqlAnalyzer.USER_DEFINED_FUNCTIONS,
                 String.join(".", createFunctionStmt.getNamePath()));
         udfBuilder.put(functionFullName, createFunctionStmt);
+      } else if (statement.nodeKind() == RESOLVED_CREATE_TABLE_FUNCTION_STMT) {
+        ResolvedCreateTableFunctionStmt createTableFunctionStmt =
+            (ResolvedCreateTableFunctionStmt) statement;
+        catalog.addTableValuedFunction(
```

Comment:
We add UDF to the catalog in SqlAnalyzer. It's probably fine to add to the catalog in either SqlAnalyzer or ZetaSQLPlannerImpl, but please keep it consistent.

Reply:
Sorry I am getting an action to address this comment.

Reply:
https://github.com/apache/beam/blob/103facbf80c2e4786e60367161ec59352196d04a/sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/SqlAnalyzer.java#L152

I suggest adding scalar functions and TVF to the catalog in the same file. So either move `catalog.add(scalar function)` to ZetaSQLPlannerImpl or move `catalog.add(TVF)` to SqlAnalyzer.

Reply:
Done.